Yoast/javascript
# Implement Dutch stemming function

## Summary

Stems Dutch words.

Word classes that are stemmed:
- noun plurals (e.g. _katten_) and diminutives (e.g. _katje_)
- all forms of adjectives (e.g. _zachte_, _zachts_, _zachter_, _zachters_, _zachtere_, _zachtst_, _zachtste_ )
- verb infinitives (e.g. _sneeuwen_) and present participles (e.g. _sneeuwend_)

Words classes that are not stemmed correctly
-  present singular (_t)_ and past simple (_-te(n), -de(n_)) verb suffixes (e.g. _sneeuwt_, _sneeuwde_, _sneeuwden_)
- past participles (e.g. _gesneeuwd_)
- word with separable prefixes 
A word like _afwerken_ should in theory get two stems: _werk_ and _afwerk_, since a form like _werkt af_ is a valid form of _afwerken_ . However, only _afwerk_ is currently returned.
- diminutive suffixes -tje (e.g. _biertje_) and -etje (e.g. _sterretje_) not always stemmed correctly (only when we can be sure that the _t/et_ is part of the suffix, not the stem). 
- Words which do not have a clear suffix are not stemmed. This includes forms of irregular words like _is_, _zijn_, _bent_... or words that follow a irregular plural formation pattern, such as words of Latin origin like _musea_. 

Words that are stemmed, but shouldn't (false positives):
- Sometimes, the characters matched by a suffix regex are part of the _stem_, e.g. the -_en_ in _tor*en*_ or the -_end_ in _week*end*_. These words will be incorrectly stemmed. (This problem will be solved in the future, either through having an exception list of words that should not get stemmed, or by adding additional suffixes to make sure each word has its stem ending attached back).

Decisions concerning cases where more than on stem is possible
- When there is a stem with both double and single vowel (e.g. maken -> mak, maak), we return the one with the _double_ vowel
- When there is a stem with both double and single consonant (e.g. paken -> pak, pakk), we return the one with the _single_ consonant
- When there is a stem with both voiced and voiceless consonant (e.g. lezen -> lez, lees), we return the one with the _voiceless_ consonant (so _s_ in this example).
- When the stem ending can be either -ieel or -iël (e.g. officieel/officiële), the stem ending in -ieel is returned.
- The -heden plural suffix is stemmed to -heid.

## Test instructions

This PR can be tested by following these steps:

* Add more words to the specs and make sure they are stemmed correctly

Fixes #298
